span.go

```diff
@@ -312,7 +312,10 @@ func setSamplingPriority(s *Span, value interface{}) bool {
 		s.context.flags = s.context.flags & (^flagSampled)
 		return true
 	}
-	if s.tracer.isDebugAllowed(s.operationName) {
+	if s.tracer.options.sampleWithoutDebugFlag {
+		s.context.flags = s.context.flags | flagSampled
+		return true
+	} else if s.tracer.isDebugAllowed(s.operationName) {
 		s.context.flags = s.context.flags | flagDebug | flagSampled
 		return true
 	}
```

Comment:
DRY it to
```go
	s.context.flags = s.context.flags | flagSampled
	if !s.tracer.options.sampleWithoutDebugFlag && s.tracer.isDebugAllowed(s.operationName) {
		s.context.flags = s.context.flags | flagDebug
	}
	return true
```

Reply:
Actually we can't, the behavior will change. Previously if the span needs to be throttled, the sampled flag won't be set. With the proposed logic, sampled flag will be set no matter debug span should be throttled or not. That means collector will see more samples, just not with debug flag.